modules/dev-utils/src/test/java/org/apache/ignite/development/utils/IgniteWalConverterTest.java

```diff
@@ -0,0 +1,381 @@
+ * limitations under the License.
+ */
+
+package org.apache.ignite.development.utils;
+
+import java.io.ByteArrayOutputStream;
+import java.io.File;
+import java.io.PrintStream;
+import java.io.RandomAccessFile;
+import java.util.Base64;
+import java.util.LinkedList;
+import java.util.List;
+import org.apache.ignite.IgniteCache;
+import org.apache.ignite.cache.CacheAtomicityMode;
+import org.apache.ignite.cache.CacheMode;
+import org.apache.ignite.configuration.CacheConfiguration;
+import org.apache.ignite.configuration.DataRegionConfiguration;
+import org.apache.ignite.configuration.DataStorageConfiguration;
+import org.apache.ignite.configuration.IgniteConfiguration;
+import org.apache.ignite.configuration.WALMode;
+import org.apache.ignite.internal.IgniteEx;
+import org.apache.ignite.internal.pagemem.wal.record.WALRecord;
+import org.apache.ignite.internal.processors.cache.persistence.wal.serializer.RecordV1Serializer;
+import org.apache.ignite.internal.util.typedef.internal.U;
+import org.apache.ignite.testframework.junits.common.GridCommonAbstractTest;
+import org.junit.Test;
+
+/**
+ * Test for IgniteWalConverter
+ * <br>
+ * Test case:
+ * <ol>
+ *     <li>
+ *     Checking utility IgniteWalConverter
+ *         <ul>
+ *             <li>Start node</li>
+ *             <li>Create cache with <a href="https://apacheignite.readme.io/docs/indexes#section-registering-indexed-types">Registering Indexed Types</a></li>
+ *             <li>Put several entity</li>
+ *             <li>Stop node</li>
+ *             <li>Read wal with specifying binaryMetadata</li>
+ *             <li>Check that the output contains all DataRecord with previously added entities</li>
+ *         </ul>
+ *     </li>
+ *     <li>
+ *     Checking utility IgniteWalConverter with out binary_meta
+ *         <ul>
+ *             <li>Start node</li>
+ *             <li>Create cache with <a href="https://apacheignite.readme.io/docs/indexes#section-registering-indexed-types">Registering Indexed Types</a></li>
+ *             <li>Put several entity</li>
+ *             <li>Stop node</li>
+ *             <li>Read wal with <b>out</b> specifying binaryMetadata</li>
+ *             <li>Check that the output contains all DataRecord and in DataRecord not empty key and value</li>
+ *         </ul>
+ *     </li>
+ *     <li>
+ *     Checking utility IgniteWalConverter on broken WAL
+ *         <ul>
+ *             <li>Start node</li>
+ *             <li>Create cache with <a href="https://apacheignite.readme.io/docs/indexes#section-registering-indexed-types">Registering Indexed Types</a></li>
+ *             <li>Put several entity</li>
+ *             <li>Stop node</li>
+ *             <li>Change byte in WAL</li>
+ *             <li>Read wal</li>
+ *             <li>Check one error when reading WAL</li>
+ *         </ul>
+ *     </li>
+ * </ol>
+ */
+public class IgniteWalConverterTest extends GridCommonAbstractTest {
```

Comment:
Please add some tests for incorrect command line input parameters. We should be sure that all parameters are checked properly, and user gets informative error message in case of mistake or typo.
Also, it's usually not necessary to do such extended description in javadoc of test class, you can describe each test case only in method javadoc.

Reply:
Checks will be added to https://ggsystems.atlassian.net/browse/GG-26099